connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedConnectCluster.java

```diff
@@ -63,24 +71,58 @@
     private static final Properties DEFAULT_BROKER_CONFIG = new Properties();
     private static final String REST_HOST_NAME = "localhost";
 
-    private final Connect[] connectCluster;
+    private final List<Connect> connectCluster;
     private final EmbeddedKafkaCluster kafkaCluster;
     private final Map<String, String> workerProps;
     private final String connectClusterName;
     private final int numBrokers;
+    private final int numInitialWorkers;
+    private final boolean failOnUngracefulShutdown;
 
-    private EmbeddedConnectCluster(String name, Map<String, String> workerProps, int numWorkers, int numBrokers, Properties brokerProps) {
+    private EmbeddedConnectCluster(String name, Map<String, String> workerProps, int numWorkers,
+                                   int numBrokers, Properties brokerProps,
+                                   boolean failOnUngracefulShutdown) {
         this.workerProps = workerProps;
         this.connectClusterName = name;
         this.numBrokers = numBrokers;
         this.kafkaCluster = new EmbeddedKafkaCluster(numBrokers, brokerProps);
-        this.connectCluster = new Connect[numWorkers];
+        this.connectCluster = new ArrayList<>();
+        this.numInitialWorkers = numWorkers;
+        this.failOnUngracefulShutdown = failOnUngracefulShutdown;
     }
 
+    /**
+     * A more graceful way to handle abnormal exit of services in integration tests.
+     */
+    public Exit.Procedure exitProcedure = (code, message) -> {
```

Comment:
Are these public so that test classes using `EmbeddedConnectCluster` can change them?

Reply:
I think reading and reusing is what I had in mind too. But yes, overriding crossed my mind too. Not a too bad tradeoff, WDYT?

Reply:
Yeah, that's a nice simple approach.